src/main/java/com/assertthat/selenium_shutterbug/utils/image/ImageProcessor.java

```diff
@@ -129,6 +131,62 @@ public static boolean imagesAreEquals(BufferedImage image1, BufferedImage image2
         return p == 0 || p <= deviation;
     }
 
+    /**
+     * Extends the functionality of imagesAreEquals, but creates a third BufferedImage and applies pixel manipulation to it.
+     * @param image1 The first image to compare
+     * @param image2 The second image to compare
+     * @param pathFileName The output path filename for the third image, if null then is ignored
+     * @param imgExtension The file extension for the output image
+     * @param deviation The upper limit of the pixel deviation for the test
+     * @return If the test passes
+     */
+    public static boolean imagesAreEquals(BufferedImage image1, BufferedImage image2, String pathFileName, String imgExtension, double deviation) {
```

Comment:
No need to pass imgExtension as a parameter. We are only using png, since any other format will cause conversion discrepancies.